frc/1. Basics/5. Motors.md

```diff
@@ -0,0 +1,118 @@
+# Motors
+
+## Terminology
+
+An actuator is defined as "a component of a machine that is responsible for moving and controlling a mechanism or system." For our purposes, we need only consider *motors* and *pneumatics*. 
+
+Our robot is essentially entirely made up of motors. All moving components are controlled by motors or pneumatics, and our job as programmers is to figure out how to convert drive team input into the robot actually doing its job. For instance, programming will make sure that certain motors move when drive team pushes buttons or joysticks on the controller.
+
+First, some clarification:
+
+- *Motor* - the actual physical part that spins
+- *Motor Controller* - facilitator that takes in and sends signal to control the motor accordingly
+
+Motors are widely used throughout most, if not all of our robot's subsystems, e.g. on the drivetrain. At the time of writing, 1257 mainly used NEO Brushless and NEO 550 motors, by REV Robotics. 
+
+<img src="img/SPARKMAX.png" width="330" height=""> <img src="img/NEOvs550.png" width="330"> 
+
+To control these motors and actually manipulate a robot mechanism, we interface with the corresponding motor controllers--this is what we do in code. The SPARK MAX controller (also by REV Robotics) is used with NEO-type motors. 
+
+## Setup
+
+Here, we'll write in the iterative programming paradigm for brevity. Examples for command-based programming will follow in the next few files.
+
+```java
+package frc.robot;
+
+import edu.wpi.first.wpilibj.*;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+public class Robot extends TimedRobot {
+
+    private final CANSparkMax motor;
+    
+    @Override
+    public void robotInit() {
+        motor = new CANSparkMax(0, MotorType.kBrushless);
+        motor.setIdleMode(IdleMode.kBrake);
+        motor.setSmartCurrentLimit(80);
+
+    }
+
+    @Override
+    public void teleopPeriodic() {
+
+    }
+}
+```
+
+We've done a couple things so far:
+
+1. Set and declared a CANSparkMax motor controller with ID 0
+2. Configured the controller to work with a Brushless motor (i.e. NEO/NEO 550)
+3. Set the motor to "brake" mode
+4. Set a current limit of 80 amps to the motor
+4. Done all of the above with the corresponding vendor dependency file and imports
+
+Read more here on [brushed vs. brushless motors](https://cordlessdrillzone.com/drill-wars/brushless-vs-brushed-motor/) and [Brake vs. Coast mode](https://www.chiefdelphi.com/t/what-is-brake-coast-mode/163649). As for current limiting, it's just good practice to set that limit as a safety, such that risk of overdrawing/breaking the motor is reduced.
```

Comment:
current limiting is not only a good practice, it's absolutely necessary on some mechanisms to avoid damage. also why is Brake vs. Coast capitalized and not brushes vs brushless